pkg/config/creds_test.go

```diff
@@ -10,102 +10,153 @@ import (
 	"testing"
 
 	"github.com/DataDog/datadog-operator/pkg/secrets"
+	
 	"github.com/stretchr/testify/assert"
 )
 
 func Test_getCredentials(t *testing.T) {
 	tests := []struct {
 		name      string
-		setupFunc func(*secrets.DummyDecryptor)
+		setupFunc func(*secrets.DummyDecryptor, *CredentialManager)
 		want      Creds
 		wantErr   bool
-		wantFunc  func(*testing.T, *secrets.DummyDecryptor)
-		resetFunc func()
+		wantFunc  func(*testing.T, *secrets.DummyDecryptor, *CredentialManager)
+		resetFunc func(*CredentialManager)
 	}{
 		{
-			name:      "creds found, no SB",
-			setupFunc: func(*secrets.DummyDecryptor) { os.Setenv("DD_API_KEY", "foo"); os.Setenv("DD_APP_KEY", "bar") },
-			want:      Creds{APIKey: "foo", AppKey: "bar"},
-			wantErr:   false,
-			wantFunc:  func(t *testing.T, d *secrets.DummyDecryptor) { d.AssertNotCalled(t, "Decrypt") },
-			resetFunc: func() { os.Unsetenv("DD_API_KEY"); os.Unsetenv("DD_APP_KEY") },
+			name: "creds found, no SB",
+			setupFunc: func(*secrets.DummyDecryptor, *CredentialManager) {
+				os.Setenv("DD_API_KEY", "foo")
+				os.Setenv("DD_APP_KEY", "bar")
+			},
+			want:    Creds{APIKey: "foo", AppKey: "bar"},
+			wantErr: false,
+			wantFunc: func(t *testing.T, d *secrets.DummyDecryptor, cm *CredentialManager) {
+				d.AssertNotCalled(t, "Decrypt")
+				cachedCreds, cached := cm.getCredsFromCache()
+				assert.True(t, cached)
+				assert.EqualValues(t, Creds{APIKey: "foo", AppKey: "bar"}, cachedCreds)
+			},
+			resetFunc: func(cm *CredentialManager) {
+				os.Unsetenv("DD_API_KEY")
+				os.Unsetenv("DD_APP_KEY")
+				cm.resetCredsCache()
```

Comment:
I think you can remove the `cm.resetCredsCache()` since each test create a new `CredentialManager` instance.